cosmos/ingestion/ingest/enrich/enrich.py

```diff
@@ -0,0 +1,189 @@
+from typing import List
+import pandas as pd
+import logging
+import os
+from tqdm import tqdm
+import glob
+import functools
+
+from dask.distributed import Client, progress
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.ERROR)
+
+
+def needed_columns_are_in_df(to_check: List, to_interrogate: List) -> bool:
+    """
+    True if all of to_check values are in to_interrogate, else False
+    """
+    if all(x in to_interrogate for x in to_check):
+        return True
+    else:
+        return False
+
+
+class Enrich:
+    """
+    Enrich class
+    handles running semantic enrichment of tables in ingestion pipeline output parquets
+    """
+    def __init__(self, scheduler_address, client=None):
+        """
+        :param scheduler_address: Address to existing Dask scheduler
+        :param client: A Dask client. Can be passed in. If None, one will be created to connect to the scheduler
+        """
+        logger.info("Initializing Enrich object")
+        self.client = client
+        if self.client is None:
+            logger.info("Setting up client")
+            self.client = Client(scheduler_address, serializers=['msgpack', 'dask'], deserializers=['msgpack', 'dask', 'pickle'])
+            logger.info(self.client)
+
+    def __del__(self):
+        """Simple client clean up"""
+        if self.client is not None:
+            self.client.close()
+
+    def enrich(self, input_path: str, output_path: str, dataset_id: str,
+               threshold: float = 0.8, spans: int = 20):
+        """
+        main method -calls all other processing methods and outputs enriched parquet file
+        :param input_path: a directory full of parquets (ingest output) to process
+        :param output_path: a directory to put the output context enriched parquets
+        :param dataset_id: ingest process dataset_id
+        :param threshold: float cut off for postprocess table detection score to process as table caption
+        :param spans: number of words each side of label to pull in as context for each table label in content text
+                if None will use regex to pull out full stop to full stop span around the table label
+        """
+
+        for pq in glob.glob(os.path.join(input_path, '*.parquet')):
```

Comment:
Looks like you're looping over all the parquet files, making sure they have three columns below, and then in `get_contexts`, requiring that they be tables? Is it kept loose so that logic for other types of objects can get added into `get_contexts` below?

Reply:
Yes, that's exactly it. The requirement for those three columns probably won't change, but I wasn't sure what the input parquets might have or not (now or in the future).